Sources/TestSupport/misc.swift

```diff
@@ -192,3 +192,27 @@ public func loadMockPackageGraph(_ packageMap: [String: PackageDescription.Packa
     }
     return try PackageGraphLoader().load(rootManifest: rootManifest, externalManifests: externalManifests, fileSystem: fs)
 }
+
+/// Temporary override environment variables
+public func withCustomEnv(_ env: [String: String], body: () throws -> ()) throws {
```

Comment:
There should probably be a note here in the function documentation that this is super-unthreadsafe (by which I mean that using this function produces side effects in the form of environment variable mutation that's visible to other threads).  This is unfortunate but doesn't seem particularly avoidable in the absence of per-thread environment variables.  But it's worth noting these side effects, I think.